### PR TITLE
[#796] Developer documentation: Migrations

### PR DESCRIPTION
**Pull request by MariusBluem**
* based on @nickvergessen's https://www.schilljs.com/2017/09/08/migrations-for-nextcloud-apps.html
* requested: https://github.com/nextcloud/documentation/issues/796

Signed-off-by: Marius Blüm <marius@lineone.io>

### CODE REVIEW

**Comment by nickvergessen** [comment]
In the meantime we also added a command to create a migration from the current info.xml: `migrations:generate-from-schema`

**Comment by MariusBluem** [comment]
> In the meantime we also added a command to create a migration from the current info.xml: `migrations:generate-from-schema`

Thanks 😅 @nickvergessen

File: developer_manual/app/migrations.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+* `migrations:generate`:
+  This is needed to create a new migration file. This takes 2 arguments,
+  first one is the `appid`, the second one should be the `version`of your
+  app as an integer. We recommend to use the major and minor digits of your apps
+  version for that. This allows you to introduce a new migration in your branch
+  for a Nextcloud version if there is already an migration path for a newer one
+  in another branch. Since you can’t change this retroactive, we recommend to
+  leave enough space in between and therefor map the numbers to 3 digits:
+  `1.0.x => 1000`, `2.34.x => 2034`, etc.
+* `migrations:generate-from-schema`: Create a migration from the current `info.xml`.
```
Comment (nickvergessen) [inline developer_manual/app/migrations.rst]: *from the old `database.xml`

**Comment by MariusBluem** [comment]
@MorrisJobke Mind to do a final review so we can get this one in?

**Comment by MorrisJobke** [comment]
stable13 b43accc6 206247d3 aa652ca4

**Pull request merged**